### PR TITLE
Add logrotate for OpenID on app1

### PR DESCRIPTION
**Pull request by tolvstaa**
As requested by [Ticket #29287](https://support.osuosl.org/Ticket/Display.html?id=29287)

### CODE REVIEW

File: recipes/app1.rb [hunk 1]
```diff
@@ -137,3 +137,10 @@
     '8082 -c config/unicorn.rb -E deployment -D'
   end
 end
+
+# Logrotate
+logrotate_app 'OpenID' do
+  path '/home/openid-production/shared/log/*'
+  frequency 'daily'
+  rotate 30
+end
```
Comment (ramereth) [inline recipes/app1.rb]: What command is going to be run to reload the daemon? Otherwise you're going to have open file handles and issues with that.

**Comment by doublej472** [comment]
I've added the copytruncate option to logrotate. This will make sure that the file handles for any logging process aren't moved or destroyed by logrotate, instead the log is copied and then cleared, while still keeping the file handles valid.

File: recipes/app1.rb [hunk 2]
```diff
@@ -137,3 +137,11 @@
     '8082 -c config/unicorn.rb -E deployment -D'
   end
 end
+
+# Logrotate
+logrotate_app 'OpenID' do
+  path '/home/openid-production/shared/log/*'
+  options ['copytruncate']
```
Comment (ramereth) [inline recipes/app1.rb]: You can change this to ``%w(copytruncate)``

**Comment by doublej472** [comment]
@ramereth I changed the array declaration and squashed all of the commits into one.

File: spec/app1_spec.rb [hunk 3]
```diff
@@ -60,4 +60,13 @@
       expect(chef_run).to create_systemd_service(s)
     end
   end
+
+  it 'should create LogRotate service for OpenID' do
+    expect(chef_run).to enable_logrotate_app('OpenID').with(
+      path: '/home/openid-production/shared/log/*',
+      options: ['copytruncate'],
```
Comment (ramereth) [inline spec/app1_spec.rb]: You can do the same thing with ``%w()`` here :)

Comment (doublej472) [inline spec/app1_spec.rb]: My bad! It should be good now.

**Comment by doublej472** [comment]
@ramereth my bad! I've fixed the array declaration (again) in app1_spec.rb

File: recipes/app1.rb [hunk 4]
```diff
@@ -137,3 +137,11 @@
     '8082 -c config/unicorn.rb -E deployment -D'
   end
 end
+
+# Logrotate
+logrotate_app 'OpenID' do
+  path '/home/openid-production/shared/log/*'
+  options %w(copytruncate)
```
Comment (jldugger) [inline recipes/app1.rb]: We should be able to send a USR1 to the PID to convince unicorn to reopen its logfile.

Comment (ramereth) [inline recipes/app1.rb]: @jldugger is that still needed with ``copytruncate``?

Comment (jldugger) [inline recipes/app1.rb]: It's an upgrade over copytruncate. Copytruncate can potentially lose a sliver of data during the rotation.

**Comment by doublej472** [comment]
@jldugger @ramereth Now this uses postrotate to reset the logs for unicorn, saving the little bit of time that might be lost due to copy truncate.

File: recipes/app1.rb [hunk 5]
```diff
@@ -137,3 +137,14 @@
...
   end
 end
+
+# Setup logrotate, also make sure that unicorn releases the file handles
+# by sending it a USR1 signal, which will cause it reopen its logs
+logrotate_app 'OpenID' do
+  path '/home/openid-production/shared/log/*'
+  options ['postrotate',
+           '/bin/kill -USR1 /home/openid-production/current/tmp/pids/unicorn.pid',
+           'endscript']
```
Comment (jldugger) [inline recipes/app1.rb]: Pretty sure logrotate supports postrotate as a command separate from an options list. Can you use that instead?

Comment (doublej472) [inline recipes/app1.rb]: What do you mean as a separate command? I can only find examples and documentation of postrotate being used in the options list.

Comment (jldugger) [inline recipes/app1.rb]: Here's one example: https://github.com/jldugger/pwnguin-cookbook/blob/master/recipes/database.rb#L102 :slightly_smiling_face:

Comment (doublej472) [inline recipes/app1.rb]: Now it makes sense, thank you!

**Comment by doublej472** [comment]
@jldugger It's changed over now to use the postrotate command

**Comment by doublej472** [comment]
@ramereth what are your thoughts on the new commits?

**Comment by doublej472** [comment]
!bump minor ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v1.11.0.  Have a nice day!

**Pull request merged**